nion/swift/Inspector.py

```diff
@@ -1984,6 +1983,64 @@ def can_merge(self, command: Undo.UndoableCommand) -> bool:
         return isinstance(command, self.__class__) and bool(self.command_id) and self.command_id == command.command_id and self.__display_item_uuid == command.__display_item_uuid
 
 
+class ScalebarOptionsHandler(Declarative.Handler):
+    def __init__(self, display_item: DisplayItem.DisplayItem, document_controller: DocumentController.DocumentController) -> None:
+        super().__init__()
+        # Ensure the property is initialized if not already set
+        if display_item.get_display_property("show_scale_info_text") is None:
+            display_item.set_display_property("show_scale_info_text", True)
+
+        self._show_info_text_model = DisplayItemDisplayPropertyCommandModel(
+            document_controller,
+            display_item,
+            "show_scale_info_text"
+        )
+        self.notify_property_changed("show_scale_info_text")
+
+        self._draw_on_right_model = DisplayItemDisplayPropertyCommandModel(
+            document_controller,
+            display_item,
+            "draw_scale_on_right"
+        )
+
+        self._background_model = DisplayItemDisplayPropertyCommandModel(
+            document_controller,
+            display_item,
+            "show_scale_background"
+        )
+
+        u = Declarative.DeclarativeUI()
+
+        self.ui_view = u.create_column(
+            u.create_row(
+                u.create_label(text=_("Scale Bar Options:"), width=120),
+                u.create_stretch()
+            ),
+            u.create_row(
+                u.create_check_box(
+                    text=_("Show Info Text"),
+                    checked="@binding(_show_info_text_model.value)"
+                ),
+                u.create_stretch()
+            ),
+            u.create_row(
+                u.create_check_box(
+                    text=_("Draw on Right"),
+                    checked="@binding(_draw_on_right_model.value)"
+                ),
+                u.create_stretch()
+            ),
+            u.create_row(
+                u.create_check_box(
+                    text=_("Show Background"),
+                    checked="@binding(_background_model.value)"
+                ),
+                u.create_stretch()
+            )
+        )
```

Comment:
This will need rework if using "position" rather than the limited "draw on right" bool value. But on this code specifically, I would add something like `spacing=4` for items in the _column_ to make it look better.